client/pool.go

```diff
@@ -228,6 +233,9 @@ func (pool *Pool) newConnectionProducer() {
 	var err error
 
 	for {
+		if atomic.LoadUint32(&pool.closed) == 1 {
```

Comment:
in addition to this, we should wait `newConnectionProducer` goroutine exit in `Close` otherwise it may open new connection after we finished closing idle connection and release the lock.

Reply:
Yes, I wiil fix this soon.